mlir/lib/Dialect/Tosa/Transforms/TosaValidation.cpp

```diff
@@ -449,6 +449,39 @@ struct TosaValidation : public tosa::impl::TosaValidationBase<TosaValidation> {
     return true;
   }
 
+  // Recursively perform a bottom-up search to determine the maximum nesting
+  // depth, starting from a specific operation and continuing up to the function
+  // or module scope. Tosa nesting_depth starts at 0 and increments by one each
+  // time a new nested `region` is encountered.
+  static void getMaxNestedDepth(Operation *op, int32_t &depth) {
+    if (isa<mlir::func::FuncOp>(op) || isa<ModuleOp>(op))
+      return;
+
+    Block *block = op->getBlock();
+    if (!block)
+      return;
+
+    Region *region = block->getParent();
+    if (!region)
+      return;
+
+    depth++;
+    getMaxNestedDepth(region->getParentOp(), depth);
```

Comment:
Curious if there's a reason to traverse the blocks/regions? I think the parent can be fetched directly from the op `op->getParentOp()`

Reply:
No. Found out they are they are equivalent. :)